errors/errors.go

```diff
@@ -18,36 +18,32 @@ package errors
 
 import (
 	"fmt"
-	"net/http"
 )
 
+// Reason is an enumeration of possible failure causes. Each Reason
+// must map to a format which is a string contains ${xxx}.
+type Reason string
+
+// Error is the basic form of an error returned by nirvana.
 type Error interface {
-	Reason() Reason
+	Message() interface{}
```

Comment:
Returning `interface{}` loses observability on what constitutes a Message. Since this is probably the most important interface in error package, I'd prefer a struct or at least another interface type.

Reply:
Since web framework need a
type Error interface {
	Message() interface{}
	Code() int
}
here, ref: https://github.com/caicloud/nirvana/blob/master/web/handler.go#L31

So this `Message()` must return interface{}.

Reply:
Having two `Error` interfaces at two different places is awful, do we have any other solution?

@walktall @kdada

Reply:
web pkg can import errors pkg.